pkgs/applications/blockchains/oxen/default.nix

```diff
@@ -0,0 +1,72 @@
+{ stdenv, lib, fetchurl, fetchFromGitHub, fetchpatch
+, cmake, pkgconfig
+, boost, openssl, unbound
+, pcsclite, readline, libsodium, hidapi
+, rapidjson
+, curl, sqlite
+, CoreData, IOKit, PCSC
+, trezorSupport ? true
+,   libusb1  ? null
+,   protobuf ? null
+,   python3  ? null
+}:
+
+assert stdenv.isDarwin -> IOKit != null;
+assert trezorSupport -> lib.all (x: x!=null) [ libusb1 protobuf python3 ];
+
+stdenv.mkDerivation rec {
+  pname = "oxen";
+  version = "9.1.3";
+
+  src = fetchFromGitHub {
+    owner = "oxen-io";
+    repo = "oxen-core";
+    rev = "v${version}";
+    sha256 = "11g3wqn0syk47yfcsdql5737kpad8skwdxhifn2yaz9zy8n3xqqb";
+    fetchSubmodules = true;
```

Comment:
There are some more vendored libraries we can remove.

Reply:
Some are forks. And others are difficult to take out. I already tried a bit.

Reply:
Hmmm, well, not great but nothing we can change.